src/main/java/org/isf/opd/service/OpdIoOperationRepositoryImpl.java

```diff
@@ -113,11 +113,86 @@ private TypedQuery<Opd> getOpdQuery(
 			);
 		}
 		predicates.add(
-				cb.between(opd.<LocalDateTime>get("date"), dateFrom.atStartOfDay(), dateTo.plusDays(1).atStartOfDay())
+				cb.between(opd.<LocalDate>get("date"), dateFrom, dateTo.plusDays(1))
 		);
 		query.where(cb.and(predicates.toArray(new Predicate[predicates.size()])));
 
 		return entityManager.createQuery(query);
 	}
 
+	@SuppressWarnings("unchecked")	
+	@Override
+	public List<Opd> findAllOpdWhereParams2(
+			Ward ward,
+			String diseaseTypeCode,
+			String diseaseCode,
+			LocalDateTime dateFrom,
+			LocalDateTime dateTo,
+			int ageFrom,
+			int ageTo,
+			char sex,
+			char newPatient,
+			int  patientCode) {
+		return getOpdQuery2(ward, diseaseTypeCode, diseaseCode, dateFrom, dateTo, ageFrom, ageTo, sex, newPatient, patientCode).getResultList();
+	}	
+
+	private TypedQuery<Opd> getOpdQuery2(
```

Comment:
Huge repetition of code, please don't create versions "2" of methods, instead contribuite to remove others if any